packages/js/src/cache/notifications-cache.ts

```diff
@@ -0,0 +1,253 @@
+import { InMemoryCache } from './in-memory-cache';
+import type { Cache } from './types';
+import type { ListNotificationsArgs, ListNotificationsResponse, Notification } from '../notifications';
+import { NotificationEvents, NotificationsEvents, NovuEventEmitter } from '../event-emitter';
+import type { NotificationFilter } from '../types';
+import { areTagsEqual, isSameFilter } from '../utils/notification-utils';
+
+const getCacheKey = ({ tags, read, archived, limit, offset, after }: ListNotificationsArgs): string => {
+  return JSON.stringify({ tags, read, archived, limit, offset, after });
```

Comment:
I'd suggest we remove null, undefined and empty-arrays keys before adding them to the cache so that we can have more predictable and easy to debug cache keys.